assets/src/edit-story/app/prepublish/guidance/media.js

```diff
@@ -0,0 +1,220 @@
+ * Otherwise return undefined.
+ *
+ * @param {element} element The element being checked
+ * @return {Guidance|undefined} The guidance object for consumption
+ */
+export function mediaElementResolution(element) {
+  switch (element.type) {
+    case 'image':
+      return imageElementResolution(element);
+    case 'video':
+      return videoElementResolution(element);
+    case 'gif':
+      return gifElementResolution(element);
+    default:
+      throw new Error('Invalid media type');
+  }
+}
+
+function videoElementResolution(element) {
+  const videoResolutionLow =
+    element.resource.sizes.full.height <= MIN_VIDEO_HEIGHT &&
+    element.resource.sizes.full.width <= MIN_VIDEO_WIDTH;
+  const videoResolutionHigh =
+    element.resource.sizes.full.height >= MAX_VIDEO_HEIGHT &&
+    element.resource.sizes.full.width >= MAX_VIDEO_WIDTH;
+
+  if (videoResolutionHigh) {
+    return {
+      type: PRE_PUBLISH_MESSAGE_TYPES.GUIDANCE,
+      elementId: element.id,
+      message: __(
+        "Video's resolution is too high to display on most mobile devices (>4k)",
+        'web-stories'
+      ),
+    };
+  }
+
+  if (videoResolutionLow) {
+    return {
+      type: PRE_PUBLISH_MESSAGE_TYPES.GUIDANCE,
+      elementId: element.id,
+      message: __('Video has low resolution', 'web-stories'),
+    };
+  }
+
+  return undefined;
+}
+
+function imageElementResolution(element) {
+  const heightResTooLow =
+    element.resource.sizes.full.height < 2 * element.height;
+  const widthResTooLow = element.resource.sizes.full.width < 2 * element.width;
+
+  if (heightResTooLow || widthResTooLow) {
+    return {
+      type: PRE_PUBLISH_MESSAGE_TYPES.GUIDANCE,
+      elementId: element.id,
+      message: __('Image has low resolution', 'web-stories'),
+    };
+  }
+  return undefined;
+}
+
+function gifElementResolution(element) {
+  // gif/output uses the MP4 video provided by the 3P Media API for displaying gifs
+  const heightResTooLow =
+    element.resource.output.sizes.mp4.full.height < 2 * element.height;
+  const widthResTooLow =
+    element.resource.output.sizes.mp4.full.width < 2 * element.width;
+
+  if (heightResTooLow || widthResTooLow) {
+    return {
+      type: PRE_PUBLISH_MESSAGE_TYPES.GUIDANCE,
+      elementId: element.id,
+      message: __('GIF has low resolution', 'web-stories'),
+    };
+  }
+  return undefined;
+}
+
+/**
+ * Check a video element's length.
+ * If the length is longer than 1 minute, return guidance. Otherwise return undefined.
+ *
+ * @param {element} element The element being checked
+ * @return {Guidance|undefined} The guidance object for consumption
+ */
+export function videoElementLength(element) {
+  if (element.resource.length > MAX_VIDEO_LENGTH_SECONDS) {
+    return {
+      type: PRE_PUBLISH_MESSAGE_TYPES.GUIDANCE,
+      elementId: element.id,
+      message: __(
+        'Video is longer than 1 minute (suggest breaking video up into multiple segments)',
+        'web-stories'
+      ),
```

Comment:
Probably safest to use `_n()` and `sprintf` from the i18n package in case these numbers ever change.

Example:

```js
const MAX_VIDEO_LENGTH_MINUTES = Math.floor(MAX_VIDEO_LENGTH_SECONDS / 60);
sprintf(
  /* translators: %d: number of minutes; */
  _n(
    'Video is longer than %d minute (suggest breaking video up into multiple segments)',
    'Video is longer than %d minutes (suggest breaking video up into multiple segments)'
    'web-stories'
  ),
  MAX_VIDEO_LENGTH_MINUTES
)
```